feat: implement user orders page

MohamedAbdelhakem17 opened the pull request:
This pull request adds the complete **Orders Page** for issue **#23**, including API integration, UI layout, and product previews.

##  Key Features

* Integrated **Get User Orders** API using React Query.
* Added order cards showing: order number, date, total, payment method, payment status, and delivery status.
* Displayed up to **3 products** per order with a **Show All / Show Less** toggle.
* Each product shows image, name, quantity, and price.
* Added reusable components for product items and status badges.
* Improved layout, spacing, and localization with next-intl.

##  Linked Issue

Closes **#23**

Commit:
feat(orders-page): setup not-found and loading pages; additionally create GET all orders API

Commit:
Merge branch 'feat/order-page' of https://github.com/MohamedAbdelhakem17/rose-app into feat/order-page

Commit:
feat(order-page): add NotFound, Error and Loading UIs to improve UX when orders are missing, network fails, or data is loading

Commit:
feat(order-page): create function to get token without need request  and add server action to fetch orders data

Commit:
feat(order-page): implement delivery status filter and create order types with mapper function for UI rendering

Commit:
feat(order-page) : create placeholder to display orders

Commit:
refactor(navbar): restructure files, add localization, and enhance session handling

- Refactored navbar files and directory structure
- Added localization to navigation links using next-intl
- Fixed spacing and layout alignment issues
- Extended NextAuth session type with custom user fields
- Added profile menu with user actions and logout
- Integrated language toggle for locale switching
- Added theme selector for light/dark mode

Commit:
refactor(auth): updated CN input component to support toggling password visibility, integrated it into the login form, and fixed styling issues in dark mode.

Commit:
refactor(auth): enhance register form with phone & gender fields and reusable layout

- Created dedicated PhoneInput component for better validation and UX
- Added GenderSelect input with dynamic translations
- Refactored RegisterForm to use the shared CNForm component for cleaner structure
- Integrated RespValue component for localized heading and link rendering
- Improved form responsiveness and accessibility

Commit:
refactor(auth): enhance login form with add reusable layout

Commit:
refactor(middleware): redirect unauthenticated users to login with callbackUrl to return them to their original private page after successful authentication

Commit:
Merge branch 'refactor/navigation-and-auth' into feat/order-page

Commit:
feat(order-page): create Accordion Header

Commit:
feat(order-page): create order card ui and connect with api

Commit:
feat(order-page): add translation and handle skeleton display

OmarQussas committed:
feat/(checkout): add coupon stepper and discount summary section

Introduce a checkout stepper UI to guide users through the order process and integrate a dedicated coupon application section within the order summary. Includes reusable components for coupon input, applied coupon list, and dynamic pricing with discount calculations. Enhances user experience by providing clear visual feedback during coupon application and real-time total updates.

Shamsmedhat commented on src/app/[locale]/(auth)/login/_components/login-form.tsx:
```diff
@@ -105,26 +95,16 @@ export default function LoginForm() {
                     type='password'
                     placeholder='********'
-                    className={`${form.formState.errors.password?.message && 'border-red-600 dark:border-soft-pink-500 focus:ring-red-600 dark:focus:ring-soft-pink-600'}`}
+                    className={`${form.formState.errors.password?.message && 'border-red-600 focus:ring-red-600'}`}
```
We use the `cn` utility whenever we need to write JavaScript logic inside a `className`.

Shamsmedhat commented on src/app/[locale]/(public)/orders/_components/order-card.tsx:
Good job on formatting the code and keeping a clean code style 👏👏

Shamsmedhat commented on src/app/[locale]/(public)/orders/_components/order-card.tsx:
```diff
@@ -0,0 +1,261 @@
+  const [showAll, setShowAll] = useState(false);
+
+  // Variables
+  const dateLocale = locale === 'ar' ? 'ar-EG' : 'en-US';
```
This can be extracted into a constant variable so it can be reused across the app.

OmarQussas commented on src/app/[locale]/(public)/orders/_components/order-card.tsx:
Great Job so far 🌟
Would help If you can modularize this component for better readability and maintainbility

Shamsmedhat commented on src/lib/apis/orders/orders.api.ts:
```diff
@@ -0,0 +1,60 @@
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+    const token = await getToken();
+
+    // If user not logged in
```
Please make sure not to leave any commented-out code when pushing your changes unless it’s intentionally kept for future use.
In that case, add a clear comment explaining why the code is commented and what it's intended for. This helps avoid confusion and keeps the codebase clean and maintainable.

Shamsmedhat commented on src/lib/apis/orders/orders.api.ts:
```diff
@@ -0,0 +1,60 @@
+      method: 'GET',
+      headers: {
+        ...REQUEST_HEADERS,
+        Authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoiNjhlZDZkZGI3ZmVlNjhhNGMyZWI5NWQ3Iiwicm9sZSI6InVzZXIiLCJpYXQiOjE3NjE0NjE3MDN9.1OTe3mr1K5hjdhvTYhX5HEyan1MH8CdLZGnhbCorXcg`,
```
Why hardcoded the token??

MohamedAbdelhakem17 replied:
**I had it hardcoded before implementing `getToken`; it was just for testing, and I intended to change it—and I did.**

OmarQussas commented on src/app/[locale]/(public)/orders/page.tsx:
Great job! 
a try catch statement around the "getOrders" function could help

Shamsmedhat commented on src/lib/apis/orders/orders.api.ts:
```diff
@@ -0,0 +1,60 @@
+    // Map products
+    const mappedPayload: MappedOrderResponse = {
+      ...payload,
+      orders: mappingOrders(payload.orders, t),
```
Why mapping over the orders here??

MohamedAbdelhakem17 replied:
Because I need to return data ready for the UI instead of raw data, and also to keep performance by processing on the server, not the client.

Shamsmedhat replied:
Looks good, but let’s keep concerns separated, make `orders.api` only fetch raw data, then map/format it in a server component before sending to the UI. This keeps logic in the server and improves performance.

Shamsmedhat reviewed (changes requested):
Nice work on the UI 👏 However, you need to display the proper UI state when there’s no data in the cart.

<img width="1217" height="374" alt="image" src="https://github.com/user-attachments/assets/5e5e3b38-cc1c-4bf3-8f7b-7ff9ec3c5193" />

Commit:
(feat): implement clear cart with integration with api

Commit:
(feat): implement add item to cart and update the quantity of item into cart

Commit:
refactor(order-page): create hook to mange data format , remove hard token , use getToken function and remove not needed comments

Commit:
(fix):solve comments

AbdelrahmanMabrouk committed:
Merge pull request #31 from MohamedAbdelhakem17/feature/cart-page

Feature/cart page

OmarQussas committed:
finialized edits

OmarQussas committed:
Merge branch 'dev' of https://github.com/MohamedAbdelhakem17/rose-app into dev

Commit:
feat: add Dialog and Dropdown Menu components for UI enhancement

- Implemented Dialog component with customizable overlay, content, header, footer, title, and description.
- Added Dropdown Menu component with support for items, checkboxes, radio buttons, labels, and separators.
- Enhanced form components with improved error handling and accessibility features.
- Introduced Phone Input component with country selection and flag display.
- Created reusable Popover and ScrollArea components for better UI interactions.
- Updated internationalization messages for Arabic and English languages.
- Refactored user notification actions to include token-based authentication.
- Removed deprecated footer navigation constants and replaced with dynamic navigation links.
- Improved middleware for authentication checks on private and auth pages.

Commit:
fix

Commit:
Merge branch 'feat/order-page' of https://github.com/MohamedAbdelhakem17/rose-app into dev
